poincare/src/expression.cpp

```diff
@@ -321,6 +321,9 @@ void Expression::defaultDeepReduceChildren(ExpressionNode::ReductionContext redu
 
 Expression Expression::defaultShallowReduce() {
   Expression result;
+  if (sSimplificationHasBeenInterrupted) {
+    result = Undefined::Builder();
+  }
   const int childrenCount = numberOfChildren();
```

Comment:
Put the following lines in a else?

Reply:
I did not want to make the code reading more difficult but you're right, it does slightly optimise the execution. Trade-off?